rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -47,6 +47,19 @@ std::string strip_parent_path(const std::string & relative_path)
 {
   return fs::path(relative_path).filename().generic_string();
 }
+
+std::string remove_active_from_filename(const std::string& relative_path)
+{
+  fs::path path(relative_path);
+  auto filename = path.filename().string();
+  auto active_pos = filename.find(".active");
+  if(active_pos != std::string::npos) {
+    filename.replace(active_pos, 7, "");
+  }
+  auto new_path = path.parent_path() / filename;
+  return new_path.string();
```

Comment:
just a suggestion, this does not block the PR.

To make it more efficient and robust, how about the following?

```suggestion
  if (relative_path.empty()) {
    return "";  // return immediately instead of further operation
  }
  
  auto active_pos = relative_path.find(".active");
  if (active_pos == std::string::npos) {
    return relative_path;  // return immediately instead of further operation
  }
  
  fs::path path(relative_path); // create path object
  auto new_path
    = path.parent_path() / path.filename().replace_extension("").string();
  return new_path.string();
```

Reply:
Initially, I started with a simple `relative_path.find(".active");` being the code simpler, but eventually I realized that that substring might be part of the name of a folder (0.01% probability?), so I ended up with this solution.

Also, I don't think performance is a concern or that the relative_path will ever be empty (can't be)